File: app/src/main/java/com/example/testshadow/ShadowView.java
package com.example.testshadow;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.graphics.drawable.BitmapDrawable;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;

import com.google.android.material.button.MaterialButton;

public class ShadowView extends MaterialButton {


    public ShadowView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        initView();
    }

    public ShadowView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initView();
    }

    private void initView() {
        setBackgroundColor(Color.GREEN);
        setCornerRadius(10);
    }

    public ShadowView(Context context) {
        super(context);
        initView();
    }



    private Paint createShadow() {

        initView();

        Paint mShadow = new Paint();

        float radius = 10.0f;
        float xOffset = 5.0f;
        float yOffset = 5.0f;
        mShadow.setColor(Color.RED);
        // color=black
        int color = 0xFF000000;
        int redColor = Color.RED;
        mShadow.setShadowLayer(radius, xOffset, yOffset, redColor);


        return mShadow;
    }

    @Override
    protected void onDraw(Canvas canvas) {
        Paint mShadow = createShadow();
        setLayerType(LAYER_TYPE_SOFTWARE, mShadow);
//        canvas.drawPaint(mShadow);


//
//        initView();
//
        super.onDraw(canvas);
//        initView();


    }

}

File: app/src/main/java/com/example/testshadow/ShadowView.kt
//package com.example.testshadow
//
//import android.R.attr.bitmap
//import android.content.Context
//import android.graphics.*
//import android.util.AttributeSet
//import com.google.android.material.button.MaterialButton
//
//
//class ShadowView : MaterialButton {
//
//    private val bgPaint: Paint = Paint()
//
//    init {
//        bgPaint.color = Color.GREEN
//    }
//
//    constructor(context: Context) : super(context) {
//        initView(context)
//    }
//
//    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
//        initView(context, attrs)
//    }
//
//    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
//        context!!,
//        attrs,
//        defStyleAttr
//    ) {
//        initView(context, attrs, defStyleAttr)
//    }
//
//
//    var mBackgroundColor = Color.RED
//    private fun initView(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) {
//        bgPaint.color = this.mBackgroundColor
//        bgPaint.isAntiAlias = true
//        bgPaint.style = Paint.Style.FILL
//
//    }
//
//    val mShadow = Paint()
//
//    override fun onDraw(canvas: Canvas?) {
//        super.onDraw(canvas)
//
//
//        canvas ?: return
//
//        mShadow.setShadowLayer(20.0f, 5.0f, 5.0f, -0x1000000)
//
//        canvas.drawPaint(mShadow)
//
//
//    }
//
//
//}
//
//
//
//
